src/app/shell/result/result.component.ts

```diff
@@ -58,17 +62,29 @@ export class ResultComponent implements OnInit, OnDestroy {
   constructor(
     private actions$: Actions,
     private store: Store,
-    public navigationBarService: NavigationBarService
+    public navigationBarService: NavigationBarService,
+    private route: ActivatedRoute,
+    private router: Router,
+
   ) { }
 
   ngOnInit(): void {
 
+    this.route.params.subscribe(params => {
+      if (params.param === 'map') {
```

Comment:
I see you have this.viewType.map, so use it instead comparing to the string  'map'